nixos/modules/security/default.nix

```diff
@@ -0,0 +1,21 @@
+{ config, lib, ... }:
+let
+  cfg = config.security;
+in
+{
+  options = {
+    security.lsm = lib.mkOption {
+      type = lib.types.listOf lib.types.str;
+      default = [ ];
+      description = ''
+        A list of the LSMs to initialize.
+      '';
+    };
+  };
+
+  config = lib.mkIf (lib.lists.length cfg.lsm > 0) {
+    boot.kernelParams = [
+      "lsm=${lib.concatStringsSep "," cfg.lsm}"
```

Comment:
What if the entries are not unique? I can imagine at least one scenario where the `security.lsm` list is implicitly concatted with, e.g., a Nixpkgs module or a 3rd party flake that tries to set it. We should be more strict here imo.

Reply:
I'm not sure what to do about that. I think the kernel chooses whatever is the first one to provide the option.

Reply:
Probably should set an assertion to blow up eval if there is a duplicate. That is an error.

Reply:
With the kernel parameters? I'm not sure if there's a reliable way to do it that also wouldn't cause infinite recursion.

Reply:
You can just use `assertions =` in the module, I think? Probably something involving lib.lists.unique.

Reply:
Ok, we've got a way to detect conflicts now.